library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -183,4 +203,35 @@ private void deleteBatchesForIds(List<Long> batchIdsToDelete, Collection<FileDow
     public Cursor retrieveFor(BatchQuery query) {
         return resolver.query(downloadsUriProvider.getBatchesUri(), null, query.getSelection(), query.getSelectionArguments(), query.getSortOrder());
     }
+
+    private static class StatusCountMap {
+
+        private final SparseArrayCompat<Integer> statusCounts = new SparseArrayCompat<>(PRIORITISED_STATUSES_SIZE);
+
+        public boolean hasNoItemsWithStatuses(List<Integer> excludedStatuses) {
+            boolean hasOtherItems = false;
+            for (int status : excludedStatuses) {
+                boolean hasItemsForStatus = hasCountFor(status);
+                if (hasItemsForStatus) {
+                    hasOtherItems = true;
+                    break;
+                }
+            }
+            return hasOtherItems;
+        }
+
+        public boolean hasCountFor(int status) {
+            return statusCounts.get(status, 0) > 0;
```

Comment:
We have the autoboxing as soon as we add something to the array. In Android OS there is a SparseIntArray which is backed by an `int[]` but there is no equivalent in the support library